great_expectations/rule_based_profiler/rule/rule.py

```diff
@@ -64,13 +64,14 @@ def generate(
             )
             self._parameters[domain.id] = parameter_container
             parameter_builder: ParameterBuilder
-            for parameter_builder in self._parameter_builders:
-                parameter_builder.build_parameters(
-                    parameter_container=parameter_container,
-                    domain=domain,
-                    variables=variables,
-                    parameters=self._parameters,
-                )
+            if self._parameter_builders:
+                for parameter_builder in self._parameter_builders:
```

Comment:
@Shinnnyshinshin We are changing to property: `self.parameter_builders` and similar for `ExpectationConfigurationBuilder` list.

Reply:
Yup this is something I'm often guilty of. Let's utilize the property (even within the class) as a general best practice.